### PR TITLE
Remove AutoUpdate functionality from Core

### PR DESCRIPTION
**Pull request by confused-Techie**
Since merging #656, we no longer have a need to keep autoupdate logic within the core of Pulsar. Especially considering how much of that logic executes at startup, at least for Windows users.

This PR removes all AutoUpdate functionality from Pulsar. That is the `atom.autoUpdate` API, the usage of this API within the core `about` package, and everywhere else it might be found.

This is a breaking change, as we are removing an entire endpoint, but doing so can follow more in line with the fact that we don't think supporting a squirrel based auto update will happen anytime soon. And I have never seen a community package that utilizes this API in the slightest. Although, if we really thought it was an issue, we could absolutely look at adding a service that can be exposed from the `pulsar-updater` package.

Beyond removal of all this logic, there is a single point of UI to consider.

The `about` package consumed the status bar to display update information, since this is no longer the case, `about` no longer consumes the status bar, and since the `about` package will no longer have any update information, I've gone ahead and hardcoded the message that displays. Where currently, any Pulsar user would see the following:

image

Our users would now see the following:

image

Where the "How to update" links to the readme of the `pulsar-updater` package, to inform users on how to update.

> Note after recent changes: This view has been updated to include a "Check Now" button, which will trigger a check via the `pulsar-updater` package for updates

---

While there was initial want to only 'mothball' this code, since it's so pervasive within the still existing codebase, it seemed best to remove it, rather than leave comments everywhere. But with that the ADR document will document what was the last commit to contain the code before removal, and will contain a link to this PR, so that we could easily restore it in the future if needed.

### CODE REVIEW

File: packages/about/lib/components/about-view.js [hunk 1]
```diff
@@ -160,12 +167,31 @@ module.exports = class AboutView extends EtchComponent {
...
+        { className: 'about-updates group-start' },
+        $.div(
+          { className: 'about-updates-box' },
+          $.div(
+            { className: 'about-updates-status' },
+            $.div(
+              { className: 'about-updates-item app-unsupported' },
+              $.span(
+                { className: 'about-updates-label is-strong' },
+                'Updates have been moved to the package `pulsar-updater`',
```
Comment (DeeDeeG) [inline packages/about/lib/components/about-view.js:180]:
```suggestion
                'Updates have been moved to the package ', $.code({}, 'pulsar-updater'), '.',
```

Style tweak?
Before:
Backticks - Updates have been moved to the package `pulsar-updater`
After:
Code element - Updates have been moved to the package pulsar-updater


This makes the line ~94 characters long, if you want me to indent it to make the line length shorter I can re-jigger it a bit.

Comment (confused-Techie) [reply packages/about/lib/components/about-view.js:180]: Ehh, this works for me, thanks!

**Review by DeeDeeG — APPROVED** [review]
I haven't pored over _all_ of the code changes in detail, but I did test the updater package and the way it's wired up on the About page, using the Cirrus binary from this PR, and it all seems to be working right for me!

(Tested on macOS 10.15 on an Intel mac.)

Tangentially: I have some more style fixes that I like the look of that I can either propose here in the next day or so, or else _happy to propose said style changes as a follow-up PR instead._

Regardless, this gets the thumbs up from me! 👍 Very cool integration on the About page IMO, and pretty good user experience with the new package IMO!

(The toast notification feels a bit spammy for me during testing on this PR's binary, which has a 1.107.x Rolling version, so I see the update notification on every window reload, but I could press a button to dismiss it, or disable the auto check in settings, or disable the pacakge, or... etc. At least it's configurable.)

Also a plus: Sayanara, dead code!

Soooo, long way of saying Approve+ on this from me.

**Comment by confused-Techie** [comment]
With approval from @DeeDeeG (Which thanks btw), and confirming on Discord that nobody has any big questions on this one, I'm gonna go ahead and merge this PR.

Although it's important that this PR is tested on platforms prior to being included within the next rolling release, so I'm going to make sure to download this rolling myself, and hope others do as well.

**Pull request merged**